File: OOP_HW6/Rectangle.cpp
//
// Created by User on 2021/6/1.
//

#include "Rectangle.h"
Rectangle::Rectangle(const string& string, const int &height,const int &width)
:Shape(string){
    cout<<"constructor in Rectangle"<<endl;
    this->height = height;
    this->width = width;
}
void Rectangle::print(){
    Shape::print();
    cout<<" Rectangle, height "<<this->height<<", width "<<this->width<<", area "<<get_area()<<endl;
}
double Rectangle::get_area() {
    return this->height*this->width;
}

File: OOP_HW6/Rectangle.java
class Rectangle extends Shape{
    private int height;
    private int width;
    public Rectangle(String string, int height, int width){
        super(string);
        this.height = height;
        this.width = width;
    }
    public void print(){
        super.print();
        System.out.println(" Rectangle, height "+this.height+", width "+this.width+", area "+get_area());
    }
    public double get_area(){
        return this.height*this.width;
    }
}